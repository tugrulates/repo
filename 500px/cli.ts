// deno-lint-ignore-file no-console
/**
 * Command-line interface for the `500px` client.
 *
 * @module cli
 */

import { Command, EnumType, ValidationError } from "@cliffy/command";
import { Table } from "@cliffy/table";
import { displayVersion } from "@roka/package/version";
import { CATEGORIES, fiveHundredPx, type Photo } from "./500px.ts";

/**
 * Run the `500px` tool with the given command-line arguments.
 *
 * @param args Command-line arguments.
 * @returns The exit code of the command.
 */
export async function cli(args: string[]): Promise<number> {
  const cmd = new Command()
    .name("500px")
    .description("Interact with 500px.")
    .usage("<command> [options]")
    .version(await displayVersion())
    .action((): void => cmd.showHelp())
    .command("discover", discoverCommand())
    .command("follows", followsCommand())
    .command("photos", photosCommand());
  try {
    await cmd.parse(args);
  } catch (e: unknown) {
    if (e instanceof ValidationError) {
      cmd.showHelp();
      console.error(`‚ùå ${e.message}`);
      return 1;
    }
    const errors = (e instanceof AggregateError) ? e.errors : [e];
    for (const error of errors) {
      console.error(`‚ùå ${error.message}`);
      if (error["cause"] && error["cause"]["error"]) {
        console.error(error.cause.error);
      }
    }
    return 2;
  }
  return 0;
}

function discoverCommand() {
  // skips photos copied from VCG
  const skip = [/^\/vcg-/];
  return new Command()
    .description("Prints a list of active and high quality users on 500px.")
    .example(
      "500px discover",
      "Prints a list of users with high scored photos.",
    )
    .example("500px discover --filter food", "Finds food photographers.")
    .example(
      "500px discover --filter macro --filter animals",
      "Either category.",
    )
    .example("500px discover --json | jq", "Query users as JSON.")
    .type("category", new EnumType(Object.values(CATEGORIES).map((c) => c.opt)))
    .option(
      "--category <category:category>",
      "Categories to filter results on.",
      { collect: true },
    )
    .option("--json", "Output the list of users as JSON.")
    .action(async ({ category, json }) => {
      const categories = Object.values(CATEGORIES).filter((c) =>
        !category || category.includes(c.opt)
      );
      const client = fiveHundredPx();
      const photos = await client.forYouFeed({ categories, limit: 1000 });
      const users = photos.map((photo) => photo.photographer.canonicalPath)
        .filter((user) => !skip.some((re) => re.test(user)));
      const result = { discover: Array.from(new Set(users)) };
      if (json) console.log(JSON.stringify(result, undefined, 2));
      else result.discover.forEach((user) => console.log(`üë§ ${user}`));
    });
}

function followsCommand() {
  return new Command()
    .description("Prints follower information on 500px.")
    .example("500px follows", "Prints follow counts.")
    .example("500px follows --follows", "Follow users who follow.")
    .example("500px follows --unfollow", "Unfollow users who dont' follow.")
    .example("500px follows --follow --unfollow", "Matches both lists.")
    .example("500px follows --json", "Outputs JSON of follower information.")
    .example(
      "500px follows --json | jq",
      "Query JSON for follower information.",
    )
    .example(
      "500px follows --json | jq '.dontFollowBack[].username'",
      "List users who are followed but don't follow back.",
    )
    .example(
      "500px follows --json | jq '.notFollowingBack[].username'",
      "List users who follow but are not followed back.",
    )
    .arguments("<username:string>")
    .option("--json", "Output the follower information as JSON.")
    .action(async ({ json }, username) => {
      const client = fiveHundredPx();
      const [following, followers] = await Promise.all([
        client.following(username),
        client.followers(username),
      ]);
      const result = {
        following,
        followers,
        dontFollowBack: following.filter(({ id }) =>
          !followers.some((user) => user.id === id)
        ),
        notFollowingBack: followers.filter(({ id }) =>
          !following.some((user) => user.id === id)
        ),
      };

      if (json) console.log(JSON.stringify(result, undefined, 2));
      else {
        console.log(`üë§ Following ${result.following.length} people.`);
        console.log(`üë§ Followed by ${result.followers.length} people.`);
      }
    });
}

function photosCommand() {
  function output(photos: Photo[]) {
    new Table()
      .body(photos.map((photo) => [
        `üèûÔ∏è ${photo.name}`,
        `üìà${photo.pulse.highest}`,
        `‚ù§Ô∏è ${photo.likedByUsers.totalCount}`,
        `üëÅÔ∏è ${photo.timesViewed}`,
      ]))
      .render();
  }
  return new Command()
    .description("Prints the list of photos for a 500px user.")
    .example("500px photos <username>", "Prints the list of photos for a user.")
    .example("500px photos <username> --json | jq", "Query photos as JSON.")
    .arguments("<username:string>")
    .option("--json", "Output the photo information as JSON.")
    .action(async ({ json }, username) => {
      const client = fiveHundredPx();
      const photos = await client.photos(username);

      if (json) console.log(JSON.stringify(photos, undefined, 2));
      else output(photos);
    });
}

if (import.meta.main) Deno.exit(await cli(Deno.args));
