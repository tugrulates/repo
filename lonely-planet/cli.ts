// deno-lint-ignore-file no-console
/**
 * Command-line interface for the `lonely-planet` client.
 *
 * @module cli
 */

import { Command } from "@cliffy/command";
import { Table } from "@cliffy/table";
import { version } from "@roka/forge/version";
import {
  type Attraction,
  type Destination,
  lonelyPlanet,
} from "./lonely-planet.ts";

/**
 * Run the `lonely-planet` tool with the given command-line arguments.
 *
 * @param args Command-line arguments.
 * @returns The exit code of the command.
 */
export async function cli(args: string[]): Promise<number> {
  const EMOJIS = {
    Continent: "🌍",
    Country: "🏳️",
    Region: "🏞️",
    City: "🏙️",
    Neighborhood: "🏘️",
    Attractions: "🎡",
    Feature: "📰",
    News: "📢",
  };
  await new Command()
    .name("lonely-planet")
    .description("Explores data from Lonely Planet.")
    .version(await version({ release: true, target: true }))
    .example("lonely-planet big sur", "Search destinations for 'big sur'.")
    .example("lonely-planet --attractions amsterdam", "Search attractions.")
    .example("lonely-planet --stories amsterdam", "Search stories.")
    .example("lonely-planet --destinations --attractions --stories", "All.")
    .example("lonely-planet --json | jq", "Stream destinations as json.")
    .arguments("[keywords...:string]")
    .option("--destinations", "Include destinations in the results.")
    .option("--attractions", "Include attractions in the results.")
    .option("--stories", "Include stories in the results.")
    .option("--json", "Output the search results as concatenated JSON.")
    .action(
      async (options, ...keywords) => {
        if (!options.destinations && !options.attractions && !options.stories) {
          options.destinations = true;
        }
        const client = lonelyPlanet();
        const rows: string[][] = [];
        if (options.destinations) {
          for await (const doc of client.destinations(keywords)) {
            if (options.json) console.log(JSON.stringify(doc));
            else rows.push([EMOJIS[doc.type], doc.title, breadcrumb(doc)]);
          }
        }
        if (options.attractions) {
          for await (const doc of client.attractions(keywords)) {
            if (options.json) console.log(JSON.stringify(doc));
            else rows.push([EMOJIS[doc.type], doc.title, breadcrumb(doc)]);
          }
        }
        if (options.stories) {
          for await (const doc of client.stories(keywords)) {
            if (options.json) console.log(JSON.stringify(doc));
            else rows.push([EMOJIS[doc.type], doc.title]);
          }
        }
        Table.from(rows).render();
      },
    ).parse(args);
  return 0;
}

function breadcrumb(document: Destination | Attraction) {
  return `[ ${
    document.breadcrumb.map((breadcrumb) => breadcrumb.title).join(" > ")
  } ]`;
}

if (import.meta.main) Deno.exit(await cli(Deno.args));
