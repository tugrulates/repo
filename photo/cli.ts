// deno-lint-ignore-file no-console
/**
 * Command-line interface for the `photo` library.
 *
 * The binary for `exiftool` is bundled. However, the binary compiled with
 * {@link https://jsr.io/@roka/forge | forge} needs `exiftool` installation.
 * For macOS, it can be installed using Homebrew with `brew install exiftool`.
 * For other systems, see the [official website](https://exiftool.org/).
 *
 * @module cli
 */

import { Command } from "@cliffy/command";
import { pooled } from "@roka/async/pool";
import { version } from "@roka/forge/version";
import { find } from "@roka/fs/find";
import { maybe } from "@roka/maybe";
import { distinct } from "@std/collections";
import { yellow } from "@std/fmt/colors";
import { dirname } from "node:path";
import { check, photo, sync } from "./photo.ts";

/** Run the `photos` tool. */
export async function cli(): Promise<number> {
  const cmd = new Command()
    .name("photos")
    .description("Manage photos.")
    .version(await version({ release: true, target: true }))
    .example("photos", "Lists all photos under current directory.")
    .example("photos [photo] --json", "Data of a photo with all variants.")
    .example("photos [photo] --sync", "Copy tags to all variants.")
    .arguments("[photos...:file]")
    .option("--sync", "Sync tags from source file to other variants.")
    .option("--json", "Output photo information as JSON.")
    .action(async (options, ...photos) => {
      const images = await Array.fromAsync(
        find([...photos], { name: "*.{jpg,jpeg}" }),
      );
      const dirs = distinct(images.map(dirname)).sort();
      for await (const p of pooled(dirs, (dir) => photo(dir))) {
        if (options.sync) await sync(p);
        if (options.json) console.log(JSON.stringify(p, null, 2));
        else {
          const warnings = check(p);
          const title = `üñº  ${p.title}`;
          if (warnings.length) {
            console.log(`${title} [${yellow(warnings.join(" "))}]`);
          } else console.log(title);
        }
      }
    });
  const { errors } = await maybe(() => cmd.parse());
  for (const error of errors ?? []) {
    console.error(`‚ùå ${error}`);
  }
  return errors ? 1 : 0;
}

if (import.meta.main) Deno.exit(await cli());
