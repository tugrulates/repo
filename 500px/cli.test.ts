// deno-lint-ignore-file no-console
import { mockFetch } from "@roka/http/testing";
import { fakeConsole } from "@roka/testing/fake";
import { assertSnapshot } from "@std/testing/snapshot";
import { cli } from "./cli.ts";

const TESTS = [
  "discover",
  "follows <username>",
  "photos <username>",
];

for (const test of TESTS) {
  Deno.test(`500px ${test}`, {
    sanitizeOps: false,
    sanitizeResources: false,
  }, async (t) => {
    using console = fakeConsole();
    using _fetch = mockFetch(t);
    const args = test.replace("<username>", "tugrulates").split(" ");
    await cli(args);
    await assertSnapshot(t, console.output({ wrap: "\n" }));
  });
}
