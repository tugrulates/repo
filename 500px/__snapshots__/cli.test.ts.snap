export const snapshot = {};

snapshot[`500px discover 1`] = `
"
👤 /parkthoven
👤 /josmangra
👤 /sergey_aleshchenko
👤 /JosuPerianes
👤 /KarolNienartowicz
👤 /drrana0207
👤 /dswindler
👤 /seferk
👤 /drjhnsn
👤 /oleg197095
👤 /Dannyell
👤 /agiroud
👤 /alain01789
👤 /MaureenMackenzie
👤 /made_in_juju_
👤 /Tijsewijs1
👤 /derhosch
👤 /desimoneroberto1984
👤 /Juergen_Daum
👤 /christianspahn
👤 /trixllukas
👤 /fa425893
👤 /andres_cabezas_martinez
👤 /PawelUchorczak
👤 /arsenyy
👤 /siogunat
👤 /Brett_G
👤 /Winfried_Kastner
👤 /hovallef
👤 /bmjesus
👤 /koneparsi
👤 /GeoJerry
👤 /calvanonicola
👤 /andrea_frediani_fotografie
👤 /iakovlev
👤 /matt_maciej
👤 /basujayanta0
👤 /clariz1975
👤 /michellenewportphotography
👤 /BeomyoungGo
👤 /koloniazawada2000
👤 /marcus171
👤 /ines_maria
👤 /jnewholm
👤 /fredericmonin24
👤 /tzypliatnikoviu
👤 /Gallifrey
👤 /ketho
👤 /narcisros
👤 /juliogonzalezdelgado
👤 /anaolgagarcia
👤 /fotograf93
👤 /nicholasl
👤 /pitsfotos
👤 /barbaraarussell
👤 /coedler
👤 /pgsdeepak
👤 /GEMINNA
👤 /titinelibouton
👤 /helseneddy
👤 /catsprague
👤 /amerikano1
👤 /IvarOK
👤 /edtsousa
👤 /HUSEYINAY
👤 /filterman23
👤 /jm_castanio
👤 /cvandolleweerd
👤 /hibooroo
👤 /aithamabez
👤 /ewwes
👤 /ruzhdiibrahimi
👤 /info267
👤 /owsinskipawel
👤 /arlievsky
👤 /vitstasov
👤 /HildaMurray
👤 /fabriziocasale
👤 /johnamm
👤 /MSR-Photoarts
👤 /domlacaze
👤 /stachjag
👤 /tompolicano
👤 /hughieoconnor
👤 /yingsbooks
👤 /kettiluisa
👤 /andreaschuh
👤 /anasbadrunasar
👤 /fedotkin
👤 /pinolino
👤 /francisfontpons
👤 /SnezanaPetrovic
👤 /DantePodenzana
👤 /gmspanek
👤 /ismailcalli
👤 /gilev6840
👤 /jurkoz
👤 /dariobarbani
👤 /Paul-XS
👤 /admiraljklee
👤 /sudip6
👤 /Jef-b
👤 /philippe6
👤 /idir25
👤 /dfaur
👤 /i_mayr
👤 /AntoniFigueras
👤 /teach50
👤 /betofranca1956
👤 /fiorita
👤 /gustavorr
👤 /kejowh
👤 /-hajosch-
👤 /hartonawa
👤 /HeikeBornemann
👤 /carlos1969
👤 /mostafijurrahmannasim
👤 /andriu-andriu
👤 /matthiasfeyrer
👤 /KARANESHEV3112
👤 /sunshinenmts
👤 /1966sucy
👤 /tsimgar
👤 /andreaspreisser
👤 /dlentwis
👤 /alicja_t
👤 /peterwtravel
👤 /Darius_Jankauskas
👤 /alikalik8010
👤 /hodge676
👤 /erst72
👤 /knotfreejc
👤 /ewald-seibel
👤 /corot2
👤 /WillemBentink
👤 /gerald_h
👤 /desertfoxalex1
👤 /sillyshen
👤 /Gustavor
👤 /BasantSoni
👤 /montantegerlando
👤 /adem_saljaj
👤 /bullseye
👤 /ahtamar
👤 /Ziozampi
👤 /bahatidaudi
👤 /mrilorlins
👤 /stefanradi
👤 /p13collado
👤 /karpis315
👤 /krzysztof_browko
👤 /ladybugnyanya
👤 /not_vincent_v_boudoir2
👤 /Saintek
👤 /beauties-and-more
👤 /eedee
👤 /YellowPuschl
👤 /jalicotfrancois
👤 /jmrosz
👤 /nasserosman2
👤 /kaauenwasser
👤 /BarbaraEs
👤 /matteofoiadelli
👤 /LSchiewe
👤 /rainermgillessen
👤 /peterboll
👤 /cagan
👤 /adripadmos
👤 /Wozniak
👤 /peholy
👤 /DominiqueDubied
👤 /incheye19711
👤 /kukusikp
👤 /xtofbored
👤 /haiku626
👤 /aboloczko
👤 /liehwanshen
👤 /henningsinielsen
👤 /amalvarezj
👤 /roberto_pazzi
👤 /shind_ph
👤 /pawelkawecki
👤 /capturedbytymel
👤 /paulkeates
👤 /raracyd
👤 /sylvainharnois
👤 /kazantzakis
👤 /drew60
👤 /vencafialka
👤 /eduardomorisso
👤 /PiliDelaCruz
👤 /MaryGreenKerr
👤 /svive1
👤 /nialloconnorhome
👤 /gaillardjeanmarc
👤 /karine_eye
👤 /hans24
👤 /jeyjey262
👤 /vegard3
👤 /trinitographer
👤 /mscl1501
👤 /AlbertLesnoy
👤 /thorsten-eilers
👤 /tomfindahl
👤 /susiecastellano
👤 /Alexander_Vinogradov
👤 /hagam62
👤 /bodro_photography
👤 /elieserbotelho2020
👤 /robertodipatrizi
👤 /milykrzyslaw
👤 /Rika
👤 /ki235araki
👤 /LeonidFedyantsev
👤 /antonwerner
👤 /bmasters1
👤 /aoxiuwy2345
👤 /Bernhardl123
👤 /vilamajor
👤 /bozenazdanowicz
👤 /fernandoglzsanz
👤 /sgpi
👤 /Christopher-Schlaf
👤 /andydauer
👤 /apani
👤 /ant-yamert
👤 /thalerst
👤 /Atha_Pana
👤 /airam_cr
👤 /RoelieSteinmann
👤 /Pedro-Leitao
👤 /isabellargerich
👤 /erichbucher
👤 /GuWu500px
👤 /LavanSJ
👤 /1jokr
👤 /heikessler
👤 /haeuserer280
👤 /josebrito2
👤 /RNavarro
👤 /dadude
👤 /vall-52
👤 /lar-ry27
👤 /jph01me8
👤 /fotopeter45
👤 /heiko_prechtl
👤 /dgt
👤 /ErikRozman
👤 /Ivan_Platonov
👤 /omer72
👤 /Novascotia_Photography
👤 /claus-dietercramer
👤 /monikakrajewska
👤 /kazutaka
👤 /KerryMoore
👤 /rusrusman557
👤 /cezar44
👤 /taketera
👤 /lichtklick
👤 /nakhym
👤 /juanjoseglezmoreno
👤 /dsukru324
👤 /Giancarlo_Tofanelli
👤 /michaelbilyk
👤 /michaelbottari
👤 /brjushkovstas
👤 /jeancadic
👤 /santrade2
👤 /NicoleBoxberger
👤 /onegirlphotographer
👤 /hamidearab
👤 /charlesgrixti
👤 /kbricknell
👤 /pjfry76
👤 /efa
👤 /shiggernackihoo
👤 /qczman
👤 /frankdobrushken
👤 /iulianraimond
👤 /johnhwang500
👤 /dx5481
👤 /macromaier
👤 /raulbordoni
👤 /golawi
👤 /msthaspiaakter
👤 /oitamooo
👤 /gillesbizien
👤 /riraroli
👤 /agnieszkazawada
👤 /streich60
👤 /dinabel
👤 /Thomas_Ruppel
👤 /Pajaasek
👤 /ivanphotographer008
👤 /PirjoStoor
👤 /morgannchauveau
👤 /Novisteel
👤 /neonman
👤 /martinlrosen
👤 /EvelynApinis
👤 /arjanssens
👤 /danilob
👤 /wierikko
👤 /kasio
👤 /photographyafzalkhan
👤 /YogeshBhandarkar
👤 /DennisBa
👤 /pierreallard
👤 /chrobacinskym
👤 /cirmasim
👤 /tm251
👤 /acrosphotography
👤 /shazzl
👤 /brsshn89
👤 /EdgardAguedo
👤 /berthedel
👤 /ire57
👤 /sinakamuraaurora
👤 /emonzonpx
👤 /scharpf
👤 /nataschfer
👤 /bodydock
👤 /davis37
👤 /yossiswisa
👤 /juergen0612
👤 /wisihoop
👤 /alevole
👤 /mikromakro
👤 /pfojt75
👤 /bina6511
👤 /maratneva
👤 /fpaul
👤 /jolynn
👤 /alainrenaudin
👤 /athanasio
👤 /patrycjusz
👤 /pauljanosi
👤 /bahutofotografia
👤 /jeffbernhardt
👤 /chwerner
👤 /koenigpk
👤 /nikkor80
👤 /Shotti
👤 /ivancolman64
👤 /CharlesLafrance
👤 /Anderson35mm
👤 /spam_block
👤 /photoart_by_unteidig
👤 /jonb4016
👤 /TamPhillis
👤 /andreasweidmann1
👤 /maaikeart
👤 /pho28305
👤 /prattkashtopsix
👤 /lorcachi
👤 /yourphotonude
👤 /airathan
👤 /gypsera
👤 /birdieslandscape
👤 /LIGHTPAWS
👤 /saarbergmann
👤 /pekalapiotr89
👤 /christiandubus
👤 /Bildervielfalt
👤 /Libor
👤 /fremersalex
👤 /herr_moser
👤 /didier_chowanek
👤 /CezaryKotarski
👤 /nklejmenova
👤 /puffchocola
👤 /gianfrancosato
👤 /nikodpua
👤 /timovandam
👤 /ThomasOhlsson
👤 /SughaBapodara
👤 /da_goodell
👤 /segundosanchez
👤 /smile-is-good
👤 /alfonso_st
👤 /Lydia_T
👤 /karatsoras
👤 /davidbowden
👤 /darmonkb
👤 /eroest
👤 /magdaindigo
👤 /hwfotografie
👤 /dalepedls
👤 /sdfmedia
👤 /RobertChromik
👤 /nerion
👤 /zbyszekmakulski
👤 /tiziana_bi
👤 /ocho56
👤 /gdigalak
👤 /abdulrazzak_alali
👤 /adrianwombwell
👤 /alainvd
👤 /casbalicki
👤 /gianpib
👤 /ivanwarhammerphoto
👤 /kaantacal
👤 /krigou
👤 /Charadm
👤 /mike_eyett
👤 /eberta58
👤 /eli5poll3
👤 /agnieszkawojcik
👤 /siniialex
👤 /trefore
👤 /jorbarbo416
👤 /55abk
👤 /richardkam
👤 /mjhpdx
👤 /BarrieSpence
👤 /R-W-Otto
👤 /paulpurser
👤 /alanshulik
👤 /paroq
👤 /erbennewitz
👤 /snowdog68
👤 /thruf
👤 /fluba
👤 /apgham
👤 /hrabecivan
👤 /NataliaMC
👤 /sacha8899
👤 /mauriziob2015
👤 /turker_ciftci
👤 /sallyaszustakowski
👤 /MarcoFelippe
👤 /fab4photo
👤 /joaocarlosalbuquerque
👤 /hansmann
👤 /adamgawron
👤 /stefanremter
👤 /claudiopoggioph
👤 /mswooden
👤 /trendkom
👤 /michele166
👤 /fahrisulak
👤 /stevechines53
👤 /thomasrange1973
👤 /idrlux
👤 /oscarfhevia
👤 /frank_di
👤 /djrm
👤 /stefsphotos
👤 /freschiroberto
👤 /helmuthenninger
👤 /happyranchu
👤 /paultstrange
👤 /oliverjenskuropka
👤 /venomweb
👤 /jkkeeley
👤 /demkina_ni
👤 /gudron-v
👤 /devban
👤 /homeagain80_house
👤 /DenisDavydoff
👤 /ilgarfazili
👤 /hansvanderburgh
👤 /miroslawschymura
👤 /ahulek
👤 /steen_han
👤 /Christian_Delvaux
👤 /kovac58
👤 /prettypop
👤 /maicolbaro
👤 /ToruKona
👤 /oksmv72
👤 /bernard_choquette
👤 /LuisBorges
👤 /DouglasRMP
👤 /annalisabi
👤 /dinhan1909
👤 /joe00064
👤 /shashinbito
👤 /danielefaccioli
👤 /tahagn6
👤 /cliks66photography
👤 /Mauritta
👤 /cro
👤 /salvoenrique
👤 /nekudzaphoto
👤 /jackbespalove
👤 /ekirilla1
👤 /laluzdeuninstante
👤 /slavasedov
👤 /mventseslav
👤 /rayrenatiphotography
👤 /muehli
👤 /CarnivalOfLight
👤 /carlotharolsen
👤 /SimonaH
👤 /moonray83
👤 /analoginspired
👤 /andreaeduardophoto
👤 /noteaburras1
👤 /pammy1140
👤 /sirenkiki
👤 /andreaskser
👤 /Disel
👤 /koneser
👤 /randywei
👤 /LuisZaf
👤 /MagdaBrzozowska
👤 /johnkng
👤 /icemanphotos
👤 /ehab1984bob
👤 /robert_pictures
👤 /claudio1948
👤 /dumitrunovac
👤 /xav71
👤 /Narrenlicht
👤 /verakubiak131
👤 /nude__art
👤 /andrei55
👤 /jeromecolombophotography
👤 /SOTOVandSOTOVA
👤 /Vicente_Concha
👤 /jerry_nowicki
👤 /Arnau_Bolet
👤 /caracurt9
👤 /mila49
👤 /piersparello
👤 /FranciscoHaro
👤 /ROPA_Photography
👤 /KeesStoof
👤 /BrianDk
👤 /El_Diacono
👤 /didierlaurancy
👤 /MarkusBr
👤 /lingariel
👤 /forrestherappy
👤 /otb4
👤 /carstenmeyerdierks
👤 /peter_jarrett
👤 /85mm_
👤 /revonoissim
👤 /danilocfaria
👤 /rkrueper
👤 /northlight1977
👤 /leinadzelgarseuc
👤 /rportraits
👤 /bradwarner
👤 /hasteehouwer
👤 /prtrt
👤 /fotografie_sd
👤 /jjserra13
👤 /cotton_suh
👤 /mariussecan
👤 /oomkewiltjer
👤 /Iphonistaz
👤 /timllynch
👤 /davelally03
👤 /ruimrvieira
👤 /all42
👤 /indeed97
👤 /stefan-f
👤 /usch24
👤 /johnhjic
👤 /rusya55musayev
👤 /nan__
👤 /bmierauklein
👤 /b54qryzj79
👤 /aenafotografie
👤 /korbel1943
👤 /128elen
👤 /cwikpawel
👤 /NinaPapiorek
👤 /kolesnikovaflora
👤 /fOto__Kraft
👤 /filnik
👤 /gkastal
👤 /Geckolino
👤 /mglinnemann
👤 /athensphototours
👤 /venordas
👤 /enricoyamada
👤 /dwaite1
👤 /luckylanza
👤 /_gaston
👤 /PatrickStrik
👤 /NunoAraujo1
👤 /juliendelaval
👤 /RusuConstantin-Andrei
👤 /MarkJones11
👤 /MakisBitos
👤 /BogdanBudnik
👤 /lolamaria
👤 /Hans_Zitzler
👤 /fredericprotat
👤 /gabrielerampi
👤 /quchen2011
👤 /Wieselblitz
👤 /kckemper2
👤 /janeellis1258
👤 /whiffle
👤 /raulcerezofotografia
👤 /jyool5
👤 /iseif
👤 /HerbertAFranke
👤 /torbenschmitz1
👤 /batorka74
👤 /alaindebrie
👤 /cabe6559
👤 /magomezmarquez
👤 /DmitriySoloduhin
👤 /boklah
👤 /wojciechlatka
👤 /angry_beluga
👤 /lr_production_
👤 /uderos
👤 /lapgil
👤 /Alforoban
👤 /natalicalim
👤 /RaviKashyap1
👤 /ChristianSanchez
👤 /skd3260
👤 /jksphotographer
👤 /nico_tabo
👤 /brian_stoddart
👤 /02sundog
👤 /oerdogmus66
👤 /antonzhilin27
👤 /RidgeWalker
👤 /msanchezergueta
👤 /asadifarhad920
👤 /eddieguiry
👤 /ela1934
👤 /aubeliavanneste
👤 /HeatherRoss
👤 /nottedilanga
👤 /NicolaPaoloemilio
👤 /rodinkfotografie
👤 /frostii77
👤 /Karin63
👤 /Structor
👤 /dynax111
👤 /paulkilleenphotography1
👤 /jjverdu
👤 /ernstfritzemeier
👤 /joana2
👤 /michaelhafner11
👤 /geysir_photo_natikk
👤 /Terry-JanFarrow
👤 /Pentaxk3
👤 /Garik65
👤 /stratosgaz
👤 /info10018
👤 /jfporfao
👤 /marysedupuy2
👤 /anouar19dz
👤 /soljaris99
👤 /j8zqw46952
👤 /cravaszz
👤 /buyanskyy
👤 /andreabarsali66
👤 /adsso
👤 /heberphoto
👤 /gokawa0803
👤 /geomi1
👤 /StreeTAccro
👤 /haopto
👤 /jukejuke1310
👤 /mcgoesel
👤 /antonio_leao
👤 /ik1wdc1
👤 /hiimken
👤 /stevehandy
👤 /BruniMayer
👤 /jnsconstable
👤 /lamoureux
👤 /pboskovski
👤 /loramulle
👤 /kesog261
👤 /amatverny
👤 /priiteinbaum
👤 /sergey_timoschuk
👤 /paparomeo
👤 /tamson66
👤 /piotrnowakfonon
👤 /maxu_ph
👤 /ufysmenc3v
👤 /donev
👤 /feal
👤 /rainier14411
👤 /bodyscape2
👤 /konrad_kali
👤 /kovalyov_foto
👤 /alfonso1979
👤 /justgin
👤 /maririri
👤 /fergilberto2012
👤 /Pavel71
👤 /domau63
👤 /giorgioq
👤 /vendigo
👤 /SidneyMartins
👤 /rolling-richi
👤 /pgu2005
👤 /jbrayone
👤 /alicecasters
👤 /devadipmen
👤 /VladimirTernovoy
👤 /JuergenGatte
👤 /GregJackson2
👤 /floriandeutsch
👤 /photobystefan
👤 /lguarini
👤 /johnsilverph
👤 /vatv59
👤 /ozkann
👤 /yanwararachman
👤 /alpcem
👤 /marcus_circus
👤 /mariowidmer
👤 /karlosch1
👤 /Gilbert13
👤 /nancyannrice
👤 /luminoso
👤 /yadhavan2002
👤 /felixvancakenberghe
👤 /annisen
👤 /JoostvanHaaren
👤 /richardlake
👤 /ceja
👤 /whirl256
👤 /guyanoshi
👤 /davidlund14
👤 /cimone
👤 /kyu9happy
👤 /ivarabei
👤 /TatyanaSavkina
👤 /etsurin
👤 /StahlKlaus
👤 /HiroakiOsonou
👤 /wrejzlik
👤 /stebu1972
👤 /walkerssk
👤 /JiriTulach
👤 /ujjwal_mukherjee
👤 /markworthen
👤 /SteveOlmstead
👤 /LeonardoFernndezLzaro
👤 /scottmliddell
👤 /francisk
👤 /ancrote
👤 /fotodieter
👤 /BrianScrivner
👤 /wushu-institut
👤 /erik321
👤 /-Aquamarine-
👤 /alainravellechapuis
👤 /litasoft
👤 /koliasan
👤 /boykopenchev
👤 /danipfister
👤 /MatthiasBechold
👤 /annegfoto
👤 /stephanepecqueux
👤 /shoyeb_007
👤 /rrrchocor
👤 /zurabsanikidze
👤 /av-peteghium
👤 /logges
👤 /SuntsovR
👤 /yuturjpd
👤 /gcipriani76
👤 /PPJ_Photos
👤 /zjpllh
"
`;

snapshot[`500px follows <username> 1`] = `
"
👤 Following 264 people.
👤 Followed by 129 people.
"
`;

snapshot[`500px photos <username> 1`] = `
"
🏞️ Serpent Spectation             📈89.8 ❤️ 82  👁️ 1384
🏞️ Dawn of Revelation             📈90.7 ❤️ 113 👁️ 1246
🏞️ Coordinating Phoenix           📈91.1 ❤️ 124 👁️ 1158
🏞️ Architectural Lasagna          📈91.9 ❤️ 169 👁️ 1211
🏞️ Big Spike Policy               📈76.2 ❤️ 132 👁️ 371 
🏞️ Soft and Sharp                 📈70.5 ❤️ 117 👁️ 368 
🏞️ Collective Memory              📈67.6 ❤️ 97  👁️ 117 
🏞️ Urban Luminescence             📈68.3 ❤️ 114 👁️ 136 
🏞️ Reflexive Dome                 📈67.6 ❤️ 104 👁️ 163 
🏞️ Enforced Silence               📈64.2 ❤️ 71  👁️ 229 
🏞️ Foraging Nectar                📈58.3 ❤️ 49  👁️ 211 
🏞️ Observed Narration             📈60.8 ❤️ 48  👁️ 143 
🏞️ Future Blues                   📈44.4 ❤️ 58  👁️ 176 
🏞️ Bridge to Serenity             📈42.9 ❤️ 46  👁️ 105 
🏞️ Greyhound Glimpse              📈47.3 ❤️ 44  👁️ 148 
🏞️ Guardian Panes                 📈63.1 ❤️ 61  👁️ 146 
🏞️ Iron Giant                     📈57.3 ❤️ 38  👁️ 130 
🏞️ Prickly Persona                📈58.9 ❤️ 40  👁️ 146 
🏞️ Poolside Pink                  📈39.8 ❤️ 30  👁️ 138 
🏞️ Tire and Throne                📈52.8 ❤️ 40  👁️ 99  
🏞️ Clouds and Steel               📈40.8 ❤️ 34  👁️ 84  
🏞️ Cosmic Encounter               📈38   ❤️ 25  👁️ 91  
🏞️ Tour de Facade                 📈37   ❤️ 22  👁️ 96  
🏞️ Imagined Breakthrough Mornings 📈31.9 ❤️ 15  👁️ 93  
🏞️ London Figaro                  📈46.2 ❤️ 34  👁️ 77  
🏞️ Submerged Cyclist              📈36.3 ❤️ 17  👁️ 92  
🏞️ Cloudcraft                     📈34.6 ❤️ 18  👁️ 86  
🏞️ Dawn of Flight                 📈41.5 ❤️ 15  👁️ 63  
🏞️ Wheel in the Sky               📈36.1 ❤️ 21  👁️ 68  
🏞️ Sphere of Perfection           📈35.7 ❤️ 18  👁️ 94  
"
`;
